app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,36 @@
+class CommentsController < ApplicationController
+
+  before_action :authenticate_user!, :except => [:index, :show]
+
+  def new
+    @image = Image.find(params[:image_id])
+    @comment = Comment.new
+  end
+
+  def create
+    @image = Image.find(params[:image_id])
+    @comment = @image.comments.build_with_user(comment_params, current_user)
+    if @comment.save
+      redirect_to images_path
+    else
+      render :new
+    end
+  end
+
+  def destroy
+    @comment = Comment.find(params[:id])
+    if current_user.id === @comment.user_id
```

Comment:
Compare instances rather than IDs. So this should be: `current_user == @comment.user`